components/aero_sensors/src/Files.c

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file Files.c
+ * @brief  Source code for filesystem interface
+ * @author Joshua Lafleur (josh.lafleur@outlook.com)
+ * @version 0.1
+ * @date 2022-07-23
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "Files.h"
+
+#include "ff.h"
+
+#include "ErrorHandler.h"
+
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+char curr_file[] = "A_run.bin";
+
+FATFS      fs    = { 0 };
+FIL        file  = { 0 };
+FS_State_E state = 0;
+
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/**
+ * @brief  Mount and open the first file
+ */
+void Files_Init(void)
+{
+    f_mount(&fs, "/", 1);
+    f_open(&file, (char*)&curr_file, FA_OPEN_ALWAYS | FA_READ | FA_WRITE);
+    state = FS_PAUSE;
+}
+
+/**
+ * @brief  Writes data to the filesystem
+ *
+ * @param buff Buffer to write from
+ * @param count Amount of bytes to write
+ */
+void Files_Write(const void* buff, uint32_t count)
+{
+    UINT bw;
+
+    if (state == FS_BUSY)
+        Error_Handler(); /**< Indicates timing error */
```

Comment:
this seems aggressive...

Reply:
yes I have this set as is currently to wait for testing. Once initial rounds of testing are done i will find all error handler references like this and clean just to track timing error